### PR TITLE
Upgrade corefx to 1.0.2-rc2-23931

### PR DESCRIPTION
**Pull request by agocke**
Split into two halves so this can be merged without breaking
Closed. There is now a stand-alone project for referencing
CoreCLR that can be updated independently of corefx. It will
need to be added to the closed solution before the refs in
the existing projects can be removed.

/cc @shyamnamboodiripad @jaredpar @dotnet/roslyn-infrastructure

### CODE REVIEW

File: src/Tools/CommonCoreClrRuntime/Class1.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace DoNotUse
```
Comment (jaredpar) [inline src/Tools/CommonCoreClrRuntime/Class1.cs]: Comment as to why we have this do note use project in general.  Maybe put it in a README.MD in this directory.

**Comment by jaredpar** [comment]
:+1:

**Comment by jaredpar** [comment]
test mac please

**Comment by agocke** [comment]
@jaredpar That looks like a real issue. Investigating...

**Comment by agocke** [comment]
Looks like there's some native assembly referenced by a couple of packages, `System.Native`, that's missing? 

@weshaggard Do you know which package is supposed to provide this? What should I look for in the project.lock.json?

**Comment by agocke** [comment]
Also, the fact this is succeeding on Windows has me worried. Is CoreCLR picking up machine assemblies?

File: src/Tools/Source/CompilerGeneratorTools/DeployCompilerGeneratorToolsRuntime/project.json [hunk 2]
```diff
@@ -2,8 +2,8 @@
   "dependencies": {
     "Microsoft.NETCore.Portable.Compatibility": "1.0.1-beta-23504",
     "Microsoft.NETCore.Platforms": "1.0.1-beta-23504",
```
Comment (weshaggard) [inline src/Tools/Source/CompilerGeneratorTools/DeployCompilerGeneratorToolsRuntime/project.json]: These other dependencies will need to be updated as well, especially the Platforms one.

File: src/Tools/CommonNetCoreReferences/project.json [hunk 3]
```diff
@@ -2,8 +2,8 @@
   "dependencies": {
     "Microsoft.NETCore.Portable.Compatibility": "1.0.1-beta-23504",
     "Microsoft.NETCore.Platforms": "1.0.1-beta-23504",
-    "Microsoft.NETCore.Runtime.CoreCLR": "1.0.1-beta-23504",
-    "Microsoft.NETCore.TestHost": "1.0.0-beta-23504",
+    "Microsoft.NETCore.Runtime.CoreCLR": "1.0.2-rc2-23925",
+    "Microsoft.NETCore.TestHost": "1.0.0-rc2-23925",
     "System.AppContext": "4.0.1-beta-23504",
```
Comment (weshaggard) [inline src/Tools/CommonNetCoreReferences/project.json]: All these prerelease packages need to be in-sync.

File: src/NuGet/Microsoft.CodeAnalysis.Common.nuspec [hunk 4]
```diff
@@ -34,6 +34,7 @@ Supported Platforms:
             <dependency id="System.Dynamic.Runtime" version="4.0.11-rc2-23910" />
```
Comment (eerhardt) [inline src/NuGet/Microsoft.CodeAnalysis.Common.nuspec]: Don't these all need to be bumped up to rc2-23925?

**Comment by weshaggard** [comment]
@agocke I added a couple comments where you need to update as well. In particular System.Native stuff gets pulled in by the newer version of Microsoft.NETCore.Platforms.

File: src/Test/DeployCoreClrTestRuntime/project.json [hunk 5]
```diff
@@ -1,6 +1,5 @@
 ﻿{
   "dependencies": {
-    "Microsoft.NETCore.TestHost": "1.0.0-beta-23504",
     "System.Runtime": "4.0.21-beta-23504",
```
Comment (eerhardt) [inline src/Test/DeployCoreClrTestRuntime/project.json]: similarly here, shouldn't all CoreFX dependencies move to the same build?

File: src/NuGet/Microsoft.CodeAnalysis.Common.nuspec (hunk 4 above)
Comment (jaredpar) [inline src/NuGet/Microsoft.CodeAnalysis.Common.nuspec]: @eerhardt i thought the design enabled us to upgrade CoreClr without upgrading all of CoreFX at the same time.

File: src/Tools/Source/CompilerGeneratorTools/DeployCompilerGeneratorToolsRuntime/project.json (hunk 2 above)
Comment (jaredpar) [inline src/Tools/Source/CompilerGeneratorTools/DeployCompilerGeneratorToolsRuntime/project.json]: @weshaggard why is that required?

File: src/NuGet/Microsoft.CodeAnalysis.Common.nuspec (hunk 4 above)
Comment (eerhardt) [inline src/NuGet/Microsoft.CodeAnalysis.Common.nuspec]: I think between actual shipped releases that might work. But pre-release builds aren't necessarily compatible with other pre-release builds. For example all the new TFM stuff came in between these 2 pre-release builds. That is not backwards compatible with older pre-release builds.

File: src/Tools/Source/CompilerGeneratorTools/DeployCompilerGeneratorToolsRuntime/project.json (hunk 2 above)
Comment (weshaggard) [inline src/Tools/Source/CompilerGeneratorTools/DeployCompilerGeneratorToolsRuntime/project.json]: For Microsoft.NETCore.Platforms in particular it is actually used to pull in the runtime specific packages and so it needs to be updated in order to get the newer shim packages (System.Native) which didn't even exist in the beta-23504 timeframe.

File: src/Tools/MSBuildToolset/project.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{
+  "version": "1.0.0-*",
```
Comment (jaredpar) [inline src/Tools/MSBuildToolset/project.json]: Please remove the `*`.  That breaks deterministic builds.

**Comment by jaredpar** [comment]
:+1: again

File: src/Scripting/CSharp/project.json [hunk 7]
```diff
@@ -5,8 +5,8 @@
     "System.Dynamic.Runtime": "4.0.10"
   },
   "frameworks": {
-    "dotnet5.4": {
-      "imports": "portable-net452"
+    "netstandard1.3": {
+      "imports": [ "portable-net452", "dotnet" ]
```
Comment (eerhardt) [inline src/Scripting/CSharp/project.json]: @weshaggard @ericstj - can you just import "dotnet" and not have a version? i.e. "dotnet5.4"

File: src/NuGet/Microsoft.CodeAnalysis.Common.nuspec [hunk 8]
```diff
@@ -34,6 +34,7 @@ Supported Platforms:
...
             <dependency id="System.Globalization" version="4.0.11-rc2-23910" />
             <dependency id="System.IO.FileSystem" version="4.0.1-rc2-23910" />
+            <dependency id="System.IO.FileSystem.Primitives" version="4.0.1-rc2-23910" />
```
Comment (eerhardt) [inline src/NuGet/Microsoft.CodeAnalysis.Common.nuspec]: Shouldn't these all be bumped to rc2-23931?

File: src/Scripting/CSharp/project.json (hunk 7 above)
Comment (weshaggard) [inline src/Scripting/CSharp/project.json]: Importing dotnet without a version is the same as dotnet0.0. We should however try to keep these aligned so we should do dotnet5.4 here which is the precursor to netstandard1.3

Comment (agocke) [inline src/Scripting/CSharp/project.json]: OK, I'll add that.

**Comment by agocke** [comment]
@weshaggard Could you double check my change to add empty ref directories to the nuspec? This will avoid pulling in unnecessary refs when using the runtime project.

File: src/NuGet/Microsoft.Net.Compilers.netcore.nuspec [hunk 9]
```diff
@@ -60,6 +60,8 @@
         </dependencies>
     </metadata>
     <files>
+        <file src="$emptyDirPath$/_._" target="ref/dotnet" />
```
Comment (weshaggard) [inline src/NuGet/Microsoft.Net.Compilers.netcore.nuspec]: The change looks correct assuming those are the right TFM's you need. What is prompting this change?

File: src/Compilers/Test/Utilities/CSharp/CSharpCompilerTestUtilities.csproj [hunk 10]
```diff
@@ -83,4 +83,9 @@
     <Import Project="..\..\..\..\..\build\Targets\VSL.Imports.targets" />
     <Import Project="..\..\..\..\..\build\Targets\Roslyn.Toolsets.Xunit.targets" />
   </ImportGroup>
-</Project>
```
Comment (tmat) [inline src/Compilers/Test/Utilities/CSharp/CSharpCompilerTestUtilities.csproj]: What's this?

File: src/NuGet/Microsoft.Net.Compilers.netcore.nuspec (hunk 9 above)
Comment (agocke) [inline src/NuGet/Microsoft.Net.Compilers.netcore.nuspec]: When testing these updates I made a test project. Adding the Net.Compilers.netcore package added the CodeAnalysis assemblies and all their references to my project. The purpose of the package was only provide compilers for runtime use, not CodeAnalysis for APIs.

File: src/Compilers/Test/Utilities/CSharp/CSharpCompilerTestUtilities.csproj (hunk 10 above)
Comment (agocke) [inline src/Compilers/Test/Utilities/CSharp/CSharpCompilerTestUtilities.csproj]: Whitespace change. I think there's a newline now.

File: src/NuGet/Microsoft.Net.Compilers.netcore.nuspec (hunk 9 above)
Comment (weshaggard) [inline src/NuGet/Microsoft.Net.Compilers.netcore.nuspec]: Was it adding the APIs that are directly in csc.exe and vbc.exe? If so then these placeholders will help there but if it is prompting the dependencies of this package into the compile graph which I suspect is the case here then these placeholders won't have any affect on that. If you wanted to block the promotion then you need the exclude=compile attribute on the dependency.

Comment (agocke) [inline src/NuGet/Microsoft.Net.Compilers.netcore.nuspec]: @weshaggard I see. I'll make that change later then -- thanks!
